File: BB.Core/Model/Characteristic.cs
using System;
using System.Collections.Generic;

namespace BB.Core.Model
{
    public class Characteristic
    {
        public Characteristic()
        {
            ProductCharacteristics = new List<ProductCharacteristic>();
            ProductCategories = new List<ProductCategory>();
            CharacteristicOptions = new List<CharacteristicOption>();
        }

        public Guid CharacteristicId { get; set; }

        public string Name { get; set; }

        public int Type => 1;

        public virtual ICollection<ProductCharacteristic> ProductCharacteristics { get; set; }

        public virtual ICollection<ProductCategory> ProductCategories { get; set; }

        public virtual ICollection<CharacteristicOption> CharacteristicOptions { get; set; }
    }
}

File: BB.Web/app/Model/Characteristic/Characteristic.ts
import pc = require("../Products/ProductCategory")
import ProductCategory = pc.ProductCategory;
import pch = require("../Products/ProductCharacteristic");
import ProductCharacteristic = pch.ProductCharacteristic;
import co = require("./CharacteristicOption");
import CharacteristicOption = co.CharacteristicOption;

export class Characteristic {
    public CharacteristicId: string;
    public Name: string;
    public Type: number;
    public ProductCharacteristics: ProductCharacteristic[];
    public ProductCategories: ProductCategory[];
    public CharacteristicOptions: CharacteristicOption[];
}